lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1353,3 +1354,29 @@ def _resolve_depenency_map(
             AnsibleError('\n'.join(error_msg_lines)),
             dep_exc,
         )
+    except CollectionDependencyInconsistentCandidate as dep_exc:
+        parents = [p for p in dep_exc.criterion.iter_parent() if p is not None]
+
+        error_msg_lines = [
+            (
+                'Failed to resolve the requested dependencies map. '
+                'Got the candidate {req.fqcn!s}:{req.ver!s} ({dep_origin!s}) '
+                'which didn\'t satisfy all of the following requirements:'.
+                format(
+                    req=dep_exc.candidate,
+                    dep_origin='direct request'
+                    if not parents else 'dependency of {parent!s}'.
+                    format(parent=','.join(parents))
```

Comment:
How about having a whitespace after the commas?
```suggestion
                    if not parents else 'dependency of {parents!s}'.
                    format(parents=', '.join(parents))
```